### ISSUE TITLE
New parameterization of vegetation surface roughness for forests

### ISSUE THREAD

**Issue by RonnyMeier** [ESCOMP/CTSM#1316]
The vegetation surface roughness (z0) of deciduous trees is close to the z0 of the ground when the trees shed their leaves. This is in contradiction to observations, which even observe an increase of z0 during the dormant phase. Therefore, we intend to introduce a new parameterization of the vegetation z0 for the forest PFTs, probably using the proposed parameterization of Nakai et al. (2008) (https://doi.org/10.1016/j.agrformet.2008.03.009). Steps to be taken:

1. Implement parameterization of Nakai et al. (already done). Could test other parameterizations later on too.
2. Use data of Hu et al. (2020) (https://doi.org/10.1016/j.agrformet.2020.107956) to tune parameters in parameterization.
3. Assess how new parameterization affects model performance.
4. Potentially do additional tuning.

**Comment by wwieder** [comment]
@RonnyMeier this looks like exciting work (sorry I missed your presentation last week).
We'd welcome these changes as on option that users can chose in CTSM simulations and we'd like to help you bring this into the CTSM code base.  To do this it seems to me like you kind of have two options here:
1. Submit a pull request (PR) with your progress on implementing the Nakai parameterization     

OR

2. You can wait until you have the Hu data to parameterize and evaluate your code modifications before opening up the PR.

Either option is fine. Once you start the PR you can start working with the software engineering team to put namelist switches around your code so we can turn on/off your new roughness parameterization. The [ctsm wiki](https://github.com/ESCOMP/CTSM/wiki/Quick-start-to-CTSM-development-with-git) has helpful suggestions for getting ready for and submitting your PR.     

Please reach out if you have questions or concerns along the way.

**Comment by wwieder** [comment]
To follow up, @RonnyMeier, you'll notice that @rgknox linked this issue to one in FATES.  The FATES team is also interested in this work and (I think) would welcome contributions to the FATES code base too.  Unfortunately, this is somewhat duplicative at this stage. BUT once you have the code up, running, and incorporated into the big-leaf OR FATES code it will hopefully be pretty straightforward to migrate it over to the other.  There may need to be additional testing and evaluation of the parameterizations used in each model, however.

**Comment by RonnyMeier** [comment]
@wwieder: Thank you for your support.  In the meantime, we got access to the data of Hu et al. I am working right now on adapting the proposed parameterization of Nakai et al. to the site data. So far, I can confirm that the z0 of forests in the current implementation is underestimated drastically, while the z0 for grasses and crops seems reasonable. I will contact you in case I have questions regarding the incroporation of the new parameterization in the code and make a PR once we are ready.

@rgknox: We could definitely consider introducing a similar parameterization to FATES. I assume FATES simulates the tree number density, which would be advantageous for the parameterization of Nakai. I propose that we finalize our development for the big-leaf version first. Once this is done, we can get in contact with you and discuss how our development could be added to FATES.

@edavin: Welcome
